Server/src/controllers/DataSetController.ts

```diff
@@ -0,0 +1,245 @@
+        let datasetId: number = +requestParam;
+        if (isNaN(datasetId)) {
+            response.status(400).json("Invalid data set ID entered");
+        }
+        else {
+            try {
+                let executionStatus = await this.dataSetService.addSavedDatasetService(userEmail, datasetId)
+                if (executionStatus[0]) { return response.status(200).json(executionStatus[1]); }
+                else { return response.status(400).json(executionStatus[1]); }
+            } catch (err) {
+                response.status(500).json(err);
+            }
+        }
+    }
+     */
+
+    /**
+     * This controller will take a request, send it to the getDataService to acquire an array 
+     * containing the unapproved data sets in the database
+     * 
+     * @param request
+     * An object containing the request information and parameters: Request 
+     * @param response 
+     * An object containing a response: Response
+     */
+    async createRequestForUnapprovedDatsets(request: Request, response: Response) {
+        try {
+            this.dataSetService = new DataSetService();
+            let requestResponse = await this.dataSetService.getUnapprovedAllDatasets()
+            return response.status(requestResponse.statusCode).json(requestResponse.message);
+        } catch (error) {
+            response.status(error.status).json(error.message);
+        }
+    }
+
+    /**
+     * This method verifies that a request has at least one valid search parameter and returns true if so
+     * and false if not.
+     * 
+     * @param request 
+     * An object containing the request information and parameters: Request
+     */
+    private validateInputData(request: Request) {
+        if (request.query.hasOwnProperty('datasetId') || request.query.hasOwnProperty('material')
+            || request.query.hasOwnProperty('year') || request.query.hasOwnProperty('categoryId')
+            || request.query.hasOwnProperty('lastName')) { return true }
+        else {
+            return false
+        }
+    }
+
+    async createRequestToFlagDataset(request: Request, response: Response) {
+        if (!request.query && !request.query.datasetId) {
+            response.status(400).json("No datasetID provided to flag dataset");
+        }
+        else {
+            let datasetIdToFlag = Number(request.query.datasetId)
+            let flaggedComments = String(request.query.flaggedComments)
+            let additionalComments = String(request.query.additionalComments)
+            try {
+                let requestResponse = await this.dataSetService.flagNewDataset(datasetIdToFlag, flaggedComments, additionalComments)
+                return response.status(requestResponse.statusCode).json(requestResponse.message);
+            } catch (error) {
+                response.status(error.status).json(error.message);
+            }
+        }
+    }
+
+    async createRequestToRejectDataset(request: Request, response: Response) {

```

Comment:
the file is quite big, I know we are rushing now a little, but next sprint lets reuse validation checks and try catch it should be pretty easy with JS, i have a few ideas ping me if interested